meta: add flaky test issue template

AshCripps opened the pull request:
Add a flaky test issue template to ensure that enough information is provided for investigation

I think this will be useful especially for newer people interested in helping out the build WG as it requires no access to report flaky tests, and is often an approach we ask people to take to help familarise themselves with our CI/test suite.

##### Checklist

- [X] documentation is changed or added
- [X] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

sam-github commented on .github/ISSUE_TEMPLATE/4-report-a-flaky-test.md:
```diff
@@ -0,0 +1,26 @@
+Platform: The platform the test is flaky on - e.g. `macos` or `linux`
+Console Output: A pasted console output from a failed CI job showing the whole failure of the test
+Build Links: Links to builds affected by the flaky test
+Failure Rate: How often does this test fail - e.g. 1 in 10 builds
+
+If possible please try to give a timeframe to the when the test started failing to assist with investigation
```
```suggestion

If any investigation has been done, please include any information found, such as how consistently the test fails, whether the failure could be reproduced locally, when the test started failing, or anything else you think is relevant.
```

bcoe commented:
@AshCripps this is topical. I'm working with a group of interns this summer, on a project related to accurately and automatically detecting flaky tests in a codebase (CC: @joyeecheung).

**Would love to see if we could use Node.js as a place to test the methods we're coming up with.**

I think developing a template for this is a great idea.

bcoe commented on .github/ISSUE_TEMPLATE/4-report-a-flaky-test.md:
```diff
@@ -0,0 +1,26 @@
+
+* **Test**:
+* **Platform**:
+* **Console Output**:
```
if it doesn't complicate the template too much, perhaps we should ask the user to use [collapsable markdown here](https://gist.github.com/joyrexus/16041f2426450e73f5df9391f7f7ae5f), so that the output doesn't dominate the issue.

sam-github replied:
If the markup is included with a `COPY THE FAIL TEST OUTPUT HERE` that could help a bunch.

bcoe commented on .github/ISSUE_TEMPLATE/4-report-a-flaky-test.md:
```diff
@@ -0,0 +1,26 @@
+* **Platform**:
+* **Console Output**:
+* **Build Links**:
+* **Failure Rate**:
```
I think we should perhaps give the person recommendations for what to fill in here, e.g., `rare`, `frequent`? not sure what you were picturing.

sam-github replied:
I don't think we should do this, see my suggestion above. I don't usually include anything like this. I might mention whether it is always or sometimes (but tests that fail always are usually caught by PRs because they always fail... resuming the PR build won't get past them and it would have to be fixed).

sam-github commented on .github/ISSUE_TEMPLATE/4-report-a-flaky-test.md:
```diff
@@ -0,0 +1,26 @@
+---
+
+<!--
+Thank you for reporting an flaky test.
+
+Please fill in as much of the template below as you're able.
```
```suggestion
Thank you for reporting a flaky test.

Flaky tests are tests that fail occaisonally in Node.js CI, but not consistently enough to block PRs from landing, or that are failing in CI jobs or test modes that are do not run for every PR. Since they are failing in CI, all the information for the template below should be available.
```

AshCripps commented:
@bcoe @sam-github Ive addressed your comments, I've tried the details block, it didn't look too great locally but need to see how it looks on GitHub.

sam-github commented:
@bcoe WDYT? I'd be tempted to leave it out, the tap output is usually not that long.

BridgeAR commented on .github/ISSUE_TEMPLATE/4-report-a-flaky-test.md:
```diff
@@ -0,0 +1,36 @@
+Thank you for reporting a flaky test.
+
+Flaky tests are tests that fail occaisonally in Node.js CI, but not consistently enough to block PRs from landing, 
+or that are failing in CI jobs or test modes that are do not run for every PR. 
```
```suggestion
or that are failing in CI jobs or test modes that are not run for every PR. 
```

BridgeAR commented on .github/ISSUE_TEMPLATE/4-report-a-flaky-test.md:
```diff
@@ -0,0 +1,36 @@
+
+Flaky tests are tests that fail occaisonally in Node.js CI, but not consistently enough to block PRs from landing, 
+or that are failing in CI jobs or test modes that are do not run for every PR. 
+Since they are failing in CI, all the information for the template below should be available.
```
We already ask for the information below and simplifying the text will hopefully help people actually reading the information. I therefore suggest to remove this sentence.

```suggestion
```

AshCripps commented:
I updated this to remove the details block as it didn't look right IMO and have instead put a `REPLACE ME` inside a code block so the output will be formatted properly

sam-github commented:
Landed in fb33b673f18
